packages/flutter_test/lib/src/all_elements.dart

```diff
@@ -23,15 +25,42 @@ Iterable<Element> collectAllElementsFrom(
   return CachingIterable<Element>(_DepthFirstChildIterator(rootElement, skipOffstage));
 }
 
+/// Provides a recursive, efficient, depth first search of an element tree.
+///
+/// [Element.visitChildren] does not guarnatee order, but does guarnatee stable
+/// order. This iterator also guarantees stable order, and iterates in a left
+/// to right order:
+///
+///       1
+///     /   \
+///    2     3
+///   / \   / \
+///  4   5 6   7
+///
+/// Will iterate in order 1, 2, 4, 5, 3, 6, 7. A ListQueue is used to allow for
```

Comment:
I was slightly confused here: if 1 is the rootElement (as the tree above makes it seem) it will in fact not iterate over 1, right?

Reply:
The root element is included in iteration

Reply:
How so? If I trace through the code, `current` will never return the root element because `current` is only assigned elements from `_stack` and the root element is never added to `_stack`, as far as I can tell...

Reply:
Ah! You are correct. I'll update this.